File: JavaScript/205. Isomorphic Strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
//     if(s.length != t.length) return false;
//     var a = {};
//     for(var i = 0;i < s.length;i++){
//         if(!a[s[i]]) a[s[i]] = t[i];
//         else if(a[s[i]] != t[i]) return false;
//     }
//     var a = {};
//     for(var i = 0;i < t.length;i++){
//         if(!a[t[i]]) a[t[i]] = s[i];
//         else if(a[t[i]] != s[i]) return false;
//     }
//     return true;

    if(s.length != t.length) return false;
    var a = {}, b = {};
    for(var i = 0;i < s.length;i++){
        if(a[s[i]] != b[t[i]]) return false;
        else{
            a[s[i]] = i;
            b[t[i]] = i;
        }
    }
};

File: Python/205. Isomorphic Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        def helper(s):
            dict_ = {}
            t = 0
            r = []
            for i in s:
                if i not in dict_:
                    dict_[i] = t
                    t += 1
                r.append(dict_[i])
            return r

        return helper(s) == helper(t)
